### ISSUE TITLE
Unable to create a working KubeClusterConfig from Blocks page

### ISSUE THREAD

**Issue by prefectcboyd** [PrefectHQ/prefect#8640]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I refreshed the page and this issue still occurred.
- [X] I checked if this issue was specific to the browser I was using by testing with a different browser.

### Bug summary

Attempting to create a KubeClusterConfig block in the UI is "successful" without error, but creating a deployment with this block fails.
It's debatable if this is a UI issue, or a Prefect issue - the UI does not validate the input , and the deployment build returns a pydantic error for the ClusterConfig.

### Reproduction

In the Prefect UI, add a new block -> selecting KubeClusterConfig.
Add the full contents of your kubeconfig file, along with the specific context-name.
Select "save" - the creation is successful.
Attempt to create a deployment using a KubernetesJob infrastructure block that references 
![image](https://user-images.githubusercontent.com/105737359/221026630-d9f8c1d1-ca26-4dfb-8689-05de5bda71d6.png)
![image](https://user-images.githubusercontent.com/105737359/221027128-735d5642-a579-422d-9d80-1531f0c03169.png)

### Error

 $ prefect deployment build ./healthcheck.py:healthcheck -n kubernetes_cluster_config -ib kubernetes-job/qwe
Found flow 'healthcheck'
Traceback (most recent call last):
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/blocks/core.py", line 723, in load
    return cls._from_block_document(block_document)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/blocks/core.py", line 578, in _from_block_document
    block = block_cls.parse_obj(block_document.data)
  File "pydantic/main.py", line 526, in pydantic.main.BaseModel.parse_obj
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/blocks/core.py", line 210, in __init__
    super().__init__(*args, **kwargs)
  File "pydantic/main.py", line 342, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for KubernetesJob
cluster_config -> config
  value is not a valid dict (type=type_error.dict)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/cli/_utilities.py", line 41, in wrapper
    return fn(*args, **kwargs)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 230, in coroutine_wrapper
    return run_async_in_new_loop(async_fn, *args, **kwargs)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 181, in run_async_in_new_loop
    return anyio.run(partial(__fn, *args, **kwargs))
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/cli/deployment.py", line 898, in build
    infrastructure = await Block.load(infra_block)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/Users/christopherboyd/virtualenvs/prefect2/lib/python3.9/site-packages/prefect/blocks/core.py", line 737, in load
    raise RuntimeError(
RuntimeError: Unable to load 'qwe' of block type None due to failed validation. To load without validation, try loading again with `validate=False`.

### Browers

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Prefect version

```Text
prefect version
Version:             2.8.2
API version:         0.8.4
Python version:      3.9.14
Git commit:          afbed19d
Built:               Fri, Feb 17, 2023 10:02 AM
OS/Arch:             darwin/arm64
Profile:             dev
Server type:         cloud
```

### Additional context

The issue can be fixed by casting the data to a json return with the following:

```
import yaml
import json

path = '/Users/christopherboyd/.kube/prefectAKS-dev.yaml'
with open(path, "r") as f:
	config_dict = json.dumps(yaml.safe_load(f))
print (config_dict)
```
The returned output can be pasted into the UI "config" section, and is then functional. 
This makes me think the line below should be similar to what I tested?
https://github.com/PrefectHQ/prefect/blob/main/src/prefect/blocks/kubernetes.py#L82

In short - if I parse my own kubeconfig, and paste the output from json.dumps(yaml.safe_load)) it works, but I don't feel like this is either clear or expected behavior for a user to create a KubernetesClusterConfig block.

**Comment by madkinsz** [comment]
@prefectcboyd want to give https://github.com/PrefectHQ/prefect/pull/8643 a try?
